src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/PrivilegesInterceptorImpl.java

```diff
@@ -170,18 +179,30 @@ public Boolean replaceKibanaIndex(final ActionRequest request, final String acti
 
         }
 
+        return CONTINUE_EVALUATION_REPLACE_RESULT;
+    }
+
+    private CreateIndexRequest checkAndCreateRequestIfAbsent(final String name) {
```

Comment:
This (in combination with the code executing the CreateIndexRequest inside OpenDistroSecurityFilter) might suffer from concurrency problems if several index requests are done in quick succession. Possibly one should give ResourceAlreadyExistsException a special treatment when handling errors.

Also, is it safe to have a CreateIndexRequest directly followed by an IndexRequest?

Reply:
I do not expect multiple concurrent requests that interact with the same multi-tenant index as such requests are originated from Kibana UI and it should mean that the same user is used by multiple sessions. I am not even sure that Kibana supports such use case. Anyway, I added handling for `ResourceAlreadyExistsException` as a safety net.

The existing implementation relies on ES to auto create index on `IndexRequest` and the difference with the new approach is that it is security plugin that submits CreateIndexRequest instead of ES submitting similar request as part of auto create.